### PR TITLE
feat: Typography ellipsis support tooltip

### PR DESCRIPTION
**Pull request by zombieJ**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Typography `ellipsis` support `tooltip` to customize tooltip when ellipsis.       |
| 🇨🇳 Chinese |      Typography `ellipsis` 属性添加 `tooltip` 支持以自定义收缩时的提示信息。     |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/typography/demo/ellipsis.md](https://github.com/ant-design/ant-design/blob/tooltip/components/typography/demo/ellipsis.md)
[View rendered components/typography/index.en-US.md](https://github.com/ant-design/ant-design/blob/tooltip/components/typography/index.en-US.md)
[View rendered components/typography/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/tooltip/components/typography/index.zh-CN.md)

### CODE REVIEW

File: components/typography/Base.tsx [hunk 1]
```diff
@@ -485,6 +486,15 @@ class Base extends React.Component<InternalBlockProps, BaseState> {
           {suffix}
         </>
       );
+
+      // If provided tooltip, we need wrap with span to let Tooltip inject events
+      if (tooltip) {
+        textNode = (
+          <Tooltip title={tooltip === true ? children : tooltip}>
```
Comment (kerm1it) [inline components/typography/Base.tsx:493]: 支持向 Tooltip 传递属性么？后面会不会有这方面的需求？

Comment (zombieJ) [reply components/typography/Base.tsx:493]: 暂时不支持，以后有需要可以通过 `isValidElement` 拓展 `tooltip`

**Comment by zombieJ** [comment]
筒子们来个 approve 哇~

cc @ant-design/ant-design-collaborators

**Review by hengkx — APPROVED** [review]

File: components/typography/Base.tsx [hunk 2]
```diff
@@ -41,13 +41,14 @@ interface EditConfig {
...
 
-interface EllipsisConfig {
+export interface EllipsisConfig {
   rows?: number;
   expandable?: boolean;
   suffix?: string;
   symbol?: React.ReactNode;
   onExpand?: React.MouseEventHandler<HTMLElement>;
   onEllipsis?: (ellipsis: boolean) => void;
+  tooltip?: React.ReactNode | true;
```
Comment (hengkx) [inline components/typography/Base.tsx:51]: ReactNode不是有 `boolean`

File: components/typography/Text.tsx [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
 }
 
 const Text: React.FC<TextProps> = ({ ellipsis, ...restProps }) => {
+  const mergedEllipsis = React.useMemo(() => {
+    if (ellipsis && typeof ellipsis === 'object') {
+      return omit(ellipsis, ['expandable', 'rows']);
+    }
+
+    return ellipsis;
+  }, [ellipsis]);
```
Comment (afc163) [inline components/typography/Text.tsx:17]: ellipsis 是 literal object 的话，这个 useMemo 基本起不到作用。

**Pull request merged**